src/components/[guild]/RolePlatforms/components/AddRoleRewardModal/components/AddPolygonIdPanel.tsx

```diff
@@ -0,0 +1,32 @@
+import { useEffect } from "react"
+import { useFieldArray, useWatch } from "react-hook-form"
+import { PlatformType, Visibility } from "types"
+
+type Props = {
+  onSuccess: () => void
+}
+
+const AddPolygonIDPanel = ({ onSuccess }: Props) => {
+  const roleVisibility: Visibility = useWatch({ name: ".visibility" })
+
+  const { append } = useFieldArray({
+    name: "rolePlatforms",
+  })
+
+  useEffect(() => {
+    append({
+      guildPlatform: {
+        platformName: "POLYGON_ID",
+        platformGuildId: PlatformType.POLYGON_ID.toString(),
```

Comment:
this won't be unique for each guild, let's use `${PlatformType.POLYGON_ID}-${guildId}` instead